### PR TITLE
fix invalid unsigned arithmetic.

### PR DESCRIPTION
**Pull request by ihsinme**
I believe your checks are not correct.
in my opinion they are equivalent to checks !=.
I suggest a simple fix.

### CODE REVIEW

**Comment by millert** [comment]
That code comes from the protoc-c project.  I don't think your change is safer as there are still many cases of unchecked "len - pref_len" throughout the code.  With your change, if somehow len < prefix_len, data will be set to NULL but len will be set to a huge value.  The existing code might try to allocate a huge amount if prefix_len > len but at least the size will be consistent (and such an allocation is likely to fail).

In short, there is an implicit assumption throughout the code that scanned_member->len >= scanned_member->length_prefix_len.  Since the prefix is part of the data and thus included in len I don't think it is possible for the prefix to be larger so the existing code is safe.
